### PR TITLE
[ADD] l10n_cy: Basic package

### PR DESCRIPTION
**Pull request by malb-odoo**
This commit will add the basic package for the Cyprus localization.

- Default Settings
- CAO
- Taxes
- Fiscal positions
- Tax Report

task: 3640438

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by AllePilli** [comment]
This commit will add the basic package for the Cyprus localization.
- Default Settings
- CAO
- Taxes
- Fiscal positions
- Tax Report

task: 3640438

**Comment by malb-odoo** [comment]
@AllePilli When you do a PR and change the commit msg you have to change the description of the commit it self :smile: 
image

File: addons/l10n_cy/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        'base_vat',
+    ],
+    'data': [
+        'data/menuitem_data.xml',
+        'data/account_tax_report_data.xml',
+    ],
+    'demo': [
+        'demo/demo_company.xml',
+    ],
+}
```
Comment (malb-odoo) [inline addons/l10n_cy/__manifest__.py:22]: Missing end of line, you have to set up you IDE to do this for you :smile:

**Review by malb-odoo — COMMENTED** [review]
Care a lot of end of file line are missing ! :smile:

File: addons/l10n_cy/data/account_tax_report_data.xml [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+                <field name="code">cy_2</field>
+                <field name="expression_ids">
+                    <record id="tax_report_line_2_amount" model="account.report.expression">
+                        <field name="label">balance</field>
+                        <field name="engine">tax_tags</field>
+                        <field name="formula">2</field>
+                    </record>
+                </field>
+            </record>
+            <record id="tax_report_line_3" model="account.report.line">
```
Comment (malb-odoo) [inline addons/l10n_cy/data/account_tax_report_data.xml:38]: Like olma said on the task you can add a hierachy_level 0, the line will have a grey background which will be more readable for the user :smile:

File: addons/l10n_cy/data/account_tax_report_data.xml [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+                    <record id="tax_report_line_11B_amount" model="account.report.expression">
+                        <field name="label">balance</field>
+                        <field name="engine">tax_tags</field>
+                        <field name="formula">11B</field>
+                    </record>
+                </field>
+            </record>
+        </field>
+    </record>
+</odoo>
```
Comment (malb-odoo) [inline addons/l10n_cy/data/account_tax_report_data.xml:168]: missing end of file line :grimacing:

File: addons/l10n_cy/data/menuitem_data.xml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<odoo>
+    <menuitem id="account_reports_cy_statements_menu" name="Cyprus" parent="account.menu_finance_reports" sequence="5" groups="account.group_account_readonly"/>
+</odoo>
```
Comment (malb-odoo) [inline addons/l10n_cy/data/menuitem_data.xml:4]: here too

File: addons/l10n_cy/data/template/account.account-cy.csv [hunk 5]
```diff
@@ -0,0 +1,170 @@
+"id","code","name","account_type","reconcile","name@tr","name@gr"
```
Comment (malb-odoo) [inline addons/l10n_cy/data/template/account.account-cy.csv:1]: You have translated in turkish and greec the COA but no translation for the tax report for example, any reason ?

File: addons/l10n_cy/data/template/account.account-cy.csv [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+"cy_8009","8009","Computers depreciation","expense_depreciation","False","Bilgisayar amortismanı","Απόσβεση υπολογιστών"
+"cy_8010","8010","Furniture and fixtures depreciation","expense_depreciation","False","Mobilya ve demirbaş amortismanı","Αποσβέσεις επίπλων και φωτιστικών"
+"cy_8011","8011","Warehouse equipment depreciation","expense_depreciation","False","Depo ekipmanı amortismanı","Απόσβεση εξοπλισμού αποθήκης"
+"cy_8012","8012","Motor vehicles depreciation","expense_depreciation","False","Motorlu taşıtların amortismanı","Απόσβεση μηχανοκίνητων οχημάτων"
+"cy_8100","8100","Bad debt write off","expense","False","Kötü borç siliniyor","Διαγραφή επισφαλών χρεών"
+"cy_8102","8102","Bad debt provision","expense","False","Şüpheli alacak karşılığı","Πρόβλεψη για επισφαλείς απαιτήσεις"
+"cy_8200","8200","Other non-operating income or expenses","expense","False","Diğer faaliyet dışı gelir veya giderler","Άλλα μη λειτουργικά έσοδα ή έξοδα"
+"cy_8300","8300","Disposals","expense","False","Bertaraflar","Διαθέσεις"
+"cy_8400","8400","Income tax expenses","expense","False","Gelir vergisi giderleri","Έξοδα φόρου εισοδήματος"
```
Comment (malb-odoo) [inline addons/l10n_cy/data/template/account.account-cy.csv:170]: here too

File: addons/l10n_cy/data/template/account.fiscal.position-cy.csv [hunk 7]
```diff
@@ -0,0 +1,51 @@
+"id","sequence","name","name@gr","auto_apply","vat_required","country_id","country_group_id","tax_ids/tax_src_id","tax_ids/tax_dest_id","account_ids/account_src_id","account_ids/account_dest_id"
```
Comment (malb-odoo) [inline addons/l10n_cy/data/template/account.fiscal.position-cy.csv:1]: here only translated in greec ?

File: addons/l10n_cy/data/template/account.fiscal.position-cy.csv [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+"","","","","","","","","VAT_S_IN_CY_3_G","VAT_S_OUT_EU","",""
+"","","","","","","","","VAT_S_IN_CY_3_S","VAT_S_OUT_EU","",""
+"","","","","","","","","VAT_P_IN_CY_19_G","VAT_P_OUT_EU_G","",""
+"","","","","","","","","VAT_P_IN_CY_19_S","VAT_P_OUT_EU_S","",""
+"","","","","","","","","VAT_P_IN_CY_9_G","VAT_P_OUT_EU_G","",""
+"","","","","","","","","VAT_P_IN_CY_9_S","VAT_P_OUT_EU_S","",""
+"","","","","","","","","VAT_P_IN_CY_5_G","VAT_P_OUT_EU_G","",""
+"","","","","","","","","VAT_P_IN_CY_5_S","VAT_P_OUT_EU_S","",""
+"","","","","","","","","VAT_P_IN_CY_3_G","VAT_P_OUT_EU_G","",""
+"","","","","","","","","VAT_P_IN_CY_3_S","VAT_P_OUT_EU_S","",""
```
Comment (malb-odoo) [inline addons/l10n_cy/data/template/account.fiscal.position-cy.csv:51]: missing end of file line again

File: addons/l10n_cy/data/template/account.tax-cy.csv [hunk 9]
```diff
@@ -0,0 +1,133 @@
+"id","sequence","description","invoice_label","name","price_include","amount","amount_type","type_tax_use","tax_group_id","repartition_line_ids/factor_percent","repartition_line_ids/repartition_type","repartition_line_ids/document_type","repartition_line_ids/tag_ids","repartition_line_ids/account_id","description@gr"
```
Comment (malb-odoo) [inline addons/l10n_cy/data/template/account.tax-cy.csv:1]: same comment for the translation

File: addons/l10n_cy/data/template/account.tax-cy.csv [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+"","","","","","","","","","","100","base","refund","-7||-11B","",""
+"","","","","","","","","","","100","tax","refund","-1||-4","cy_2202",""
+"VAT_S_IN_EU_19_S_ESSP","32","VAT rate for sales of services in EU with reverse charge","19% EU RC","19% EU RC","False","19.0","percent","sale","tax_group_19","100","base","invoice","+8B","","Συντελεστής ΦΠΑ για πωλήσεις υπηρεσιών στην ΕΕ με αντίστροφη χρέωση"
+"","","","","","","","","","","100","tax","invoice","","cy_2201",""
+"","","","","","","","","","","100","base","refund","-8B","",""
+"","","","","","","","","","","100","tax","refund","","cy_2201",""
+"VAT_P_IN_EU_19_S_ESSP","33","VAT rate for purchases of services in EU with reverse charge","19% EU RC","19% EU RC","False","19.0","percent","purchase","tax_group_19","100","base","invoice","+6||+7||+11B","","Συντελεστής ΦΠΑ για αγορές υπηρεσιών στην ΕΕ με αντίστροφη χρέωση"
+"","","","","","","","","","","100","tax","invoice","+1||+4","cy_2202",""
+"","","","","","","","","","","100","base","refund","-6||-7||-11B","",""
+"","","","","","","","","","","100","tax","refund","-1||-4","cy_2202",""
```
Comment (malb-odoo) [inline addons/l10n_cy/data/template/account.tax-cy.csv:133]: here too

File: addons/l10n_cy/data/template/account.tax.group-cy.csv [hunk 11]
```diff
@@ -0,0 +1,6 @@
+"id","name","country_id"
+"tax_group_19","VAT 19%","base.cy"
+"tax_group_9","VAT 9%","base.cy"
+"tax_group_5","VAT 5%","base.cy"
+"tax_group_3","VAT 3%","base.cy"
+"tax_group_0","VAT 0%","base.cy"
```
Comment (malb-odoo) [inline addons/l10n_cy/data/template/account.tax.group-cy.csv:6]: here too and also missing translation if needed ?

File: addons/l10n_cy/demo/demo_company.xml [hunk 12]
```diff
@@ -0,0 +1,41 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<odoo>
+    <record id="partner_demo_company_cy" model="res.partner">
+        <field name="name">CY Company</field>
+        <field name="vat">CY10259033P</field>
+        <field name="street">1 Agiou Dometriou</field>
+        <field name="city">Nicosia</field>
+        <field name="country_id" ref="base.cy"/>
+
```
Comment (malb-odoo) [inline addons/l10n_cy/demo/demo_company.xml:9]: could suppress this empty line

**Review by malb-odoo — COMMENTED** [review]
Thanks for the PR and good job on this ! Few comments on the translation since some file are translated and some are not ? Also missing a lot of end of file line, don't know if you have set up you IDE to do it for you anyway you should add them to avoid other people to break the github history :smile:

**Review by malb-odoo — COMMENTED** [review]
This LGTM nice work ! Still some issues on the enterprise PR to be done, also i think you should translate the balance sheet and p&l too to be consistent

**Comment by qdp-odoo** [comment]
@robodoo r+ 
@AllePilli well done :)

**Pull request closed without merging**